applications/luci-app-wireguard/luasrc/view/wireguard.htm

```diff
@@ -169,100 +79,117 @@
-        '<strong><%:Public Key%>: </strong>%s' +
-        '<br /><strong><%:Listening Port%>: </strong>%<%:s%>',
-        iface['public_key'],
-        iface['listening_port']
-      );
-      document.getElementById(ifid + "info").innerHTML = s;
-      for (var j = 0, jlen = iface['peers'].length; j < jlen; j++) {
-        var peer = iface['peers'][j];
-        var pid = ifid + peer['public_key'] + "_";
-        s = String.format(
+    for (var key in data) {
+      if (!data.hasOwnProperty(key)) { continue; }
+      var ifname = key;
+      var iface = data[key];
+      var s = "";
+      if (iface.public_key == '(none)') {
+        s += '<em><%:Interface not operational!%></em>';
+      } else {
+        s += String.format(
           '<strong><%:Public Key%>: </strong>%s',
-          peer['public_key']
+          iface.public_key
         );
-        if (peer['endpoint'] != '(none)') {
+      }
+      if (iface.listen_port > 0) {
+        s += String.format(
+          '<br /><strong><%:Listen Port%>: </strong>%<%:s%>',
+          iface.listen_port
+        );
+      }
+      if (iface.fwmark != 'off') {
+        s += String.format(
+          '<br /><strong><%:Firewall Mark%>: </strong>%<%:s%>',
+          iface.fwmark
+        );
+      }
+      document.getElementById(ifname + "_info").innerHTML = s;
+      for (var i = 0, ilen = iface.peers.length; i < ilen; i++) {
+        var peer = iface.peers[i];
+        if (peer.public_key == '(none)') {
+          s = '<em><%:Peer not operational!%></em>';
+        } else {
+          s = String.format(
+            '<strong><%:Public Key%>: </strong>%s',
+            peer.public_key
+          );
+        }
+        if (peer.endpoint != '(none)') {
           s += String.format(
             '<br /><strong><%:Endpoint%>: </strong>%s',
-            peer['endpoint']
+            peer.endpoint
           );
         }
-        if (peer['allowed_ips'].length > 0) {
+        if (peer.allowed_ips.length > 0 && peer.allowed_ips[0] != '(none)') {
           s += '<br /><strong><%:Allowed IPs%>:</strong>';
-          for (var k = 0, klen = peer['allowed_ips'].length; k < klen; k++) {
-            s += '<br />&nbsp;&nbsp;&bull;&nbsp;' + peer['allowed_ips'][k];
+          for (var k = 0, klen = peer.allowed_ips.length; k < klen; k++) {
+            s += '<br />&nbsp;&nbsp;&bull;&nbsp;' + peer.allowed_ips[k];
           }
         }
-        if (peer['persistent_keepalive'] != 'off') {
+        if (peer.persistent_keepalive != 'off') {
           s += String.format(
             '<br /><strong><%:Persistent Keepalive%>: </strong>%ss',
-            peer['persistent_keepalive']
+            peer.persistent_keepalive
           );
         }
         var icon = '<img src="<%=resource%>/icons/tunnel_disabled.png" />';
-        if (((now.getTime() / 1000) - peer['latest_handshake']) < 140) {
+        var now = new Date();
+        if (((now.getTime() / 1000) - peer.latest_handshake) < 140) {
```

Comment:
I think it's useful to show the time of the most recent handshake always, since it shows the last time a peer did anything. This is interesting.

However, you can show it in a different color, or put a "not-connected" badge or something, if the 140 second mark is reached.